plugins/amqp_trx_plugin/amqp_trx_plugin.cpp

```diff
@@ -77,7 +134,11 @@ struct amqp_trx_plugin_impl : std::enable_shared_from_this<amqp_trx_plugin_impl>
 
       trx_queue_ptr->push( trx,
                 [my=shared_from_this(), delivery_tag, trx](const fc::static_variant<fc::exception_ptr, chain::transaction_trace_ptr>& result) {
-            my->amqp_trx->ack( delivery_tag );
+            if( my->acked == ack_mode::executed ) {
+               my->amqp_trx->ack( delivery_tag );
+            } else if ( my->acked == ack_mode::in_block ) {
```

Comment:
If the transaction failed, does it make sense to ack it with the block, when it isn't in the block?

Reply:
Good point, should ack it on execute if it failed.